boards/seeed/xiao_rp2040/board.cmake

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# This configuration allows selecting what debug adapter debugging to use
+# for the Xiao RP2040 by a command-line argument.
+# It is mainly intended to support both the 'picoprobe' and 'raspberrypi-swd'
+# adapter described in "Getting started with Raspberry Pi Pico".
```

Comment:
when copying code or text, then it's important to understand what you're copying.

You mention 'picoprobe' and 'raspberrypi-swd' but in code below you are setting `cmsis-dap` which is not mentioned here.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/raspberrypi/rpi_pico/board.cmake#L16

Apologies, this was the original pico related comments. I was copying it to make the board process easier as some things and since I didn't need to modify anything here I didn't add any additional comments about it.

Reply:
I see the Sparkfun RP2040 doesn't list the other forms actually: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/sparkfun/pro_micro_rp2040/doc/index.rst#programming-and-debugging I'm not really sure if there's a consistent way this is handled currently. Thinking it may be best for me to just remove the other alternatives for now until someone can come back and test it given this.

Reply:
I've gone ahead and removed the other forms of flashing I can't test myself that I pulled in from those other RP2040 boards. Will defer to others augmenting this in the future who have the tooling to further test it.